var/spack/repos/builtin/packages/libdeflate/package.py

```diff
@@ -6,19 +6,37 @@
 from spack.package import *
 
 
-class Libdeflate(MakefilePackage):
+class Libdeflate(MakefilePackage, CMakePackage):
     """Heavily optimized library for DEFLATE/zlib/gzip compression and decompression"""
 
     homepage = "https://github.com/ebiggers/libdeflate"
     url = "https://github.com/ebiggers/libdeflate/archive/v1.7.tar.gz"
 
     maintainers("dorton21")
 
+    version("1.18", sha256="225d982bcaf553221c76726358d2ea139bb34913180b20823c782cede060affd")
+    version("1.14", sha256="89e7df898c37c3427b0f39aadcf733731321a278771d20fc553f92da8d4808ac")
     version("1.10", sha256="5c1f75c285cd87202226f4de49985dcb75732f527eefba2b3ddd70a8865f2533")
     version("1.7", sha256="a5e6a0a9ab69f40f0f59332106532ca76918977a974e7004977a9498e3f11350")
 
+    build_system(
+        conditional("makefile", when="@:1.14"),
+        conditional("cmake", when="@1.15:"),
+        default="cmake",
+    )
+
     depends_on("zlib")
     depends_on("gzip")
 
-    def patch(self):
-        filter_file(r"\/usr\/local", self.prefix, "Makefile")
+    with when("build_system=cmake"):
+        depends_on("cmake@3.7:", type="build")
+
+
+class MakefileBuilder(spack.build_systems.makefile.MakefileBuilder):
```

Comment:
We need to import `spack.build_system`[^1].  To keep it more readable you can:
```python
from spack.build_systems import cmake, makefile
```
and then use `cmake.CMakeBuilder` etc. in the class definition.
[^1]: I'm always puzzled by Python's ability to mask errors like this "sometimes"